adyenv6b2ccheckoutaddon/acceleratoraddon/web/src/com/adyen/v6/controllers/pages/AdyenComponentController.java

```diff
@@ -75,50 +74,49 @@
     @Resource(name = "acceleratorCheckoutFacade")
     private AcceleratorCheckoutFacade checkoutFacade;
 
-    @RequestMapping(value = "/payment", method = RequestMethod.POST)
-    @ResponseBody
-    public String componentPayment(final Model model,
-                                final HttpServletRequest request,
-                                @Valid final AdyenPaymentForm adyenPaymentForm,
-                                final BindingResult bindingResult) throws AdyenComponentException {
-        LOGGER.debug("Component paymentForm: " + adyenPaymentForm);
-
-        if(!isValidateSessionCart())
-        {
-            if (adyenPaymentForm.getBillingAddress() != null) {
-                adyenPaymentForm.resetFormExceptBillingAddress();
-            }
-            throw new AdyenComponentException("checkout.error.paymentethod.formentry.invalid");
-        }
+    @Resource(name = "siteBaseUrlResolutionService")
+    private SiteBaseUrlResolutionService siteBaseUrlResolutionService;
 
-        //Save payment information
-        getAdyenCheckoutFacade().handlePaymentForm(adyenPaymentForm, bindingResult);
-        if (bindingResult.hasGlobalErrors() || bindingResult.hasErrors()) {
-            LOGGER.debug(bindingResult.getAllErrors().stream().map(DefaultMessageSourceResolvable::getCode).reduce((x, y) -> (x = x + y)));
-            if (adyenPaymentForm.getBillingAddress() != null) {
-                adyenPaymentForm.resetFormExceptBillingAddress();
-            }
-            throw new AdyenComponentException("checkout.error.paymentethod.formentry.invalid");
-        }
+    @Resource(name = "baseSiteService")
+    private BaseSiteService baseSiteService;
 
+    @RequestMapping(value = "/payment", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public String componentPayment(final HttpServletRequest request) throws AdyenComponentException {
         try {
-            validateOrderForm();
+            String requestJsonString = IOUtils.toString(request.getInputStream(), String.valueOf(StandardCharsets.UTF_8));
+            JsonObject requestJson = new JsonParser().parse(requestJsonString).getAsJsonObject();
 
-            //Make payment request
             Gson gson = new GsonBuilder().disableHtmlEscaping().create();
-            String payPalDetailsJson = adyenPaymentForm.getComponentData();
-            PayPalDetails payPalDetails = gson.fromJson(payPalDetailsJson, PayPalDetails.class);
+            Boolean termsCheck = gson.fromJson(requestJson.get("termsCheck"), Boolean.class);
```

Comment:
We should move this to validateOrderForm() method since this is part of validation.

Reply:
Done